src/us/kbase/test/auth2/lib/storage/mongo/MongoStorageStartUpTest.java

```diff
@@ -151,284 +148,252 @@ public void checkCollectionNames() throws Exception {
+						.append("name", "dispcan_1"),
 				new Document("v", indexVer)
 						.append("key", new Document("_id", 1))
-						.append("name", "_id_")
-						.append("ns", "test_mongostorage.users"),
+						.append("name", "_id_"),
 				new Document("v", indexVer)
-						.append("unique", true)	
+						.append("unique", true)
 						.append("key", new Document("idents.id", 1))
 						.append("name", "idents.id_1")
-						.append("ns", "test_mongostorage.users")
 						.append("sparse", true),
 				new Document("v", indexVer)
 						.append("key", new Document("roles", 1))
 						.append("name", "roles_1")
-						.append("ns", "test_mongostorage.users")
 						.append("sparse", true),
 				new Document("v", indexVer)
 						.append("unique", true)
 						.append("key", new Document("user", 1))
-						.append("name", "user_1")
-						.append("ns", "test_mongostorage.users"),
+						.append("name", "user_1"),
 				new Document("v", indexVer)
 						.append("key", new Document("anonid", 1))
 						.append("name", "anonid_1")
-						.append("ns", "test_mongostorage.users")
 						.append("sparse", true)
 				)));
 	}
-	
+
 	@Test
 	public void indexesTestUsers() {
 		final Set<Document> indexes = new HashSet<>();
-		db.getCollection("test_users").listIndexes().forEach((Consumer<Document>) indexes::add);
+		updateIndexes("test_users", indexes);
 		assertThat("incorrect indexes", indexes, is(set(
 				new Document("v", indexVer)
 						.append("key", new Document("custrls", 1))
 						.append("name", "custrls_1")
-						.append("ns", "test_mongostorage.test_users")
 						.append("sparse", true),
 				new Document("v", indexVer)
 						.append("key", new Document("dispcan", 1))
-						.append("name", "dispcan_1")
-						.append("ns", "test_mongostorage.test_users"),
+						.append("name", "dispcan_1"),
 				new Document("v", indexVer)
 						.append("key", new Document("_id", 1))
-						.append("name", "_id_")
-						.append("ns", "test_mongostorage.test_users"),
+						.append("name", "_id_"),
 				new Document("v", indexVer)
 						.append("key", new Document("roles", 1))
 						.append("name", "roles_1")
-						.append("ns", "test_mongostorage.test_users")
 						.append("sparse", true),
 				new Document("v", indexVer)
 						.append("unique", true)
 						.append("key", new Document("user", 1))
-						.append("name", "user_1")
-						.append("ns", "test_mongostorage.test_users"),
+						.append("name", "user_1"),
 				new Document("v", indexVer)
 						.append("key", new Document("expires", 1))
 						.append("name", "expires_1")
-						.append("ns", "test_mongostorage.test_users")
 						.append("expireAfterSeconds", 0L)
 				)));
 	}
+
+	public void updateIndexes(final String name, final Set<Document> indexes) {
+		for (Document index: db.getCollection(name).listIndexes()) {
+			index.remove("ns");
+			if (index.containsKey("expireAfterSeconds")) {
+				index.put("expireAfterSeconds", Long.valueOf(index.get("expireAfterSeconds").toString()));
```

Comment:
```suggestion
                // some versions of Mongo return ints, some longs. Convert all to longs.
				index.put("expireAfterSeconds", (long) index.get("expireAfterSeconds"));
```
I think just a cast should work

Reply:
<img width="1440" alt="Screen Shot 2024-01-17 at 4 23 47 PM" src="https://github.com/kbase/auth2/assets/22202096/cbe5a692-6317-4962-ad07-b5fb97d7f8f1">

Reply:
Interesting, I figured autoboxing would handle than. In that case `Long.valueOf(...)` should work

Reply:
still need to cast to string before calling Long.valueOf

Reply:
I don't understand, you're calling toString on some class, not casting. What the heck is the type of expireAfterSeconds?

Reply:
Well, it's an Integer from the error above

Reply:
Try https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#longValue--

Having to convert to a string and back is crazy, there's got to be some way of converting without extraneous steps

Reply:
index here is a Document
index.get method return an Object
That's why need to convert to string before calling Long.valueOf

Reply:
https://mongodb.github.io/mongo-java-driver/4.11/apidocs/bson/org/bson/Document.html#getInteger(java.lang.Object)

Reply:
Or cast to Integer or Number and then call longValue()

Reply:
👍

Reply:
:+1: